### PR TITLE
simpler package structure and renamings

### PR DESCRIPTION
**Pull request by passsy**
I looked at the dependencies of all classes and the packages before where too tight coupled, that's why I moved them into `core`. The only replaceable component is the storage implementation (currently done with a content provider)

- merged packages `accessor`, `migration` and `storage` into `core`
- `TrayStorage` (former `ModularizedStorage`) is now abstract and part of the `core`. All possible storage implementations have to extend `TrayStorage`
- the default implementation of `TrayStorage` is now `ContentProviderStorage` which was previously called `TrayStorage`. Could lead to some confusion.
- `ContentProviderStorage` and all dependencies belong into the `provider` package
- `ModularizedTrayPreference` is now called `AbstractTrayPreference`

### Before vs After
<div align="center">
  
  
</div>

### CODE REVIEW

File: library/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <application>
 
         <provider
-            android:name="net.grandcentrix.tray.provider.TrayProvider"
+            android:name=".provider.TrayContentProvider"
```
Comment (albrecht87) [inline library/src/main/AndroidManifest.xml]: Du wirst auf Android 4.0.3 Probleme bekommen, wenn du nicht den vollständigen Pfad angibst.

File: library/src/androidTest/java/net/grandcentrix/tray/provider/TrayProviderHelperTest.java [hunk 2]
```diff
@@ -54,6 +58,40 @@
...
+        matrixCursor.addRow(new Object[]{
+                "key",
+                "value",
+                "module",
+                created.getTime(),
+                updated.getTime(),
+                "migratedKey"
+        });
+        assertTrue(TextUtils.isEmpty(""));
+        assertFalse(TextUtils.isEmpty("asdf"));
```
Comment (thaussma) [inline library/src/androidTest/java/net/grandcentrix/tray/provider/TrayProviderHelperTest.java]: Werden hier die TextUtils getestet?

Comment (passsy) [inline library/src/androidTest/java/net/grandcentrix/tray/provider/TrayProviderHelperTest.java]: I assume the [TextUtils](http://developer.android.com/reference/android/text/TextUtils.html) to work correctly

Comment (passsy) [inline library/src/androidTest/java/net/grandcentrix/tray/provider/TrayProviderHelperTest.java]: Oh, just saw it. Dumb asserts

File: library/src/main/java/net/grandcentrix/tray/provider/TrayProviderHelper.java [hunk 3]
```diff
@@ -161,4 +164,30 @@ public void wipe() {
         clear();
         mContext.getContentResolver().delete(mTrayUri.getInternal(), null, null);
     }
+
+    /**
+     * converts a {@link Cursor} to a {@link TrayItem}
+     * <p>
+     * This is not a secondary constructor in {@link TrayItem} because the columns are a
+     * implementation detail of the provider package
```
Comment (passsy) [inline library/src/main/java/net/grandcentrix/tray/provider/TrayProviderHelper.java]: docu wrong

**Pull request merged**